flocker/control/_protocol.py

```diff
@@ -229,16 +230,16 @@ def _send_state_to_connections(self, connections):
                                     configuration=configuration,
                                     state=state):
             for connection in connections:
-                with LOG_SEND_TO_AGENT(
-                        self.logger, agent=connection) as action:
-                    connection.callRemote(
+                action = LOG_SEND_TO_AGENT(self.logger, agent=connection)
+                with action.context():
+                    d = DeferredContext(connection.callRemote(
```

Comment:
Sort of unrelated but does `repr(connection)` actually tell anyone anything useful?  I'm guessing this ends up looking something like `<ControlAMP at 0x7fe094b57650>`... Can you file an issue for turning this field into something useful?  Probably by logging the node identifier this connection is associated with, I'd guess.